### PR TITLE
new tool: HITRAN database

### PR DESCRIPTION
**Pull request by keflavich**
WIP: so far only a reader is implemented.

The HITRAN database (https://www.cfa.harvard.edu/hitran/) includes molecular line parameters and is more extensive than splatalogue, in particular including infrared transitions.

### CODE REVIEW

**Comment by migueldvb** [comment]
This is working well. Any chance to have this PR merged?

**Comment by keflavich** [comment]
Yeah, but it would be best to have tests.  I don't even remember writing this code, but blame says I did...

**Comment by bsipocz** [comment]
Can't agree more with @keflavich, tests would be nice, and at least a very basic narrative docs page to show the usage of it.

**Comment by migueldvb** [comment]
I will add some tests for the read_hitran_file function.

**Comment by keflavich** [comment]
Once this passes, since we now have at least basic docs and tests, I'm happy to merge it.  @migueldvb, let me know if there's more you'd like to add first.

**Comment by keflavich** [comment]
Oh, we should add:

 * [ ] a changelog entry
 * [ ] an entry in `index.rst`

File: docs/hitran/hitran.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+************************************
+HITRAN Queries (`astroquery.hitran`)
+************************************
+
+Getting started
+===============
+
+This module provides an interface to the `HITRAN`_ database API.  It can
+download a data file including transitions for a particular molecule in a given
+wavenumber range.  The file is downloaded in the `~hitran.cache_location`
```
Comment (bsipocz) [inline docs/hitran/hitran.rst]: You need to use the full path to make sphinx resolve it as an API link, e.g. ``~astroquery.hitran.cache_location``. Warnings for unresolvable links cause the build to fail.

File: docs/hitran/hitran.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+directory and can be opened with a reader function that returns a table of
+spectral lines including all accessible parameters.
+
+Examples
+========
+
+This will download all transitions of the main isotopologue of water between
+the wavenumbers of 3400 and 4100 cm\ :sup:`-1`\ .
```
Comment (bsipocz) [inline docs/hitran/hitran.rst]: Does this work for you locally? Singe backticks are special for sphinx, probably you will need to use double backticks here

Comment (migueldvb) [inline docs/hitran/hitran.rst]: This is working for me with sphinx 1.4.8

**Comment by migueldvb** [comment]
There is a travis error that should not be related with these chnages.
 ``Error: Cannot remove astropy::aplpy-1.1.1-py35_0.tar.bz2 becaue it is pinned. Use --no-pin to override``

**Comment by keflavich** [comment]
[This](https://travis-ci.org/astropy/astroquery/jobs/173733788#L578) should probably affect other affiliated packages too.  If we can get this fixed (or confirm that it's not something we should fix within astroquery), I think this is ready to merge.

**Comment by bsipocz** [comment]
@keflavich - This is a weird issue (haven't seen it with other packages yet), and definitely has nothing to do with this PR.

**Comment by keflavich** [comment]
@bsipocz OK, thanks for raising the aplpy issue.  I think we'll mark this as 'xfail' and merge if we can't find a solution quickly, but with a note that we should unmark xfail once we figure out what's going on.

**Comment by bsipocz** [comment]
I probably have a workaround, but still waiting for travis to pass on it. I can hit the merge button on this if you prefer, or you can also add the failing build to the ``allow_failures:`` section in ``.travis.yml``.

**Comment by bsipocz** [comment]
Travis passed, so I'm merging this.

**Comment by bsipocz** [comment]
@keflavich - Any particular reason for this module not following the common class API? Noticed while trying to get rid of the usage of commons.send_request() in favour of self._request...

**Comment by keflavich** [comment]
@bsipocz no.  probably just used the outdated template.

**Pull request merged**